emission/analysis/modelling/trip_model/models.py

```diff
@@ -0,0 +1,1194 @@
+########################################################################
+## Copied from /e-mission-eval-private-data/TRB_label_assist/models.py##
+########################################################################
+
```

Comment:
Is this merged over with commit history? Does not appear to be

Reply:
Currently, its Not. I was unaware of this way of moving files. 
I'll do it now.

Reply:
I did this and was able to move models.py file with its commit history. However, the `models.py` has its dependency on `clustering.py` and `data_wrangling.py` file in `TRB_label_assist`. 

I can similarly copy `data_wrangling.py`( with its commit history) file since it doesn't seem to have any further dependencies.

However, `clustering.py` has further dependencies on other files. But the functions ( from the `clustering.py`) required by `models.py` are `get_distance_matrix` and `single_cluster_purity` . None of these two functions have further dependencies . So I can  copy the clustering.py ( along with the commit history) to e-mission-server and remove the unnecessary bit.

@shankari Let me know if this strategy work

Reply:
ok, so `models.py` was *not* a direct copy from `TRB_label_assist`, you edited it further?
I don't think you should copy `clustering.py` over - you should refactor it to pull out the two functions that are dependencies and copy only the file with the pulled out code.
However, note that you have pending changes to `clustering.py` 
https://github.com/e-mission/e-mission-eval-private-data/pull/37

You should probably resolve those first, and get that PR merged before refactoring.

Reply:
Note that I am also fine with importing the bulk of the python files implementing the algorithm in here. And just having the notebooks and the evaluation code in there

```
performance_eval.py
regenerate_classification_performance_results.py
```

This is particularly important if we want to retrain or reevaluate the model periodically.

Reply:
>ok, so models.py was not a direct copy from TRB_label_assist, you edited it further?

Yes, In the previous version (when I simply dumped the code)  I removed all the classes not required by the RF model. I merged 'clustering.py' with the relevant class there and moved  a few functions  in 'utils.py' here in `e-mission-server`.

  In the current form, `model.py`( with its commit history) is the same as in `eval-private-data`  with no changes.
  
> However, note that you have pending changes to `clustering.py` [e-mission/e-mission-eval-private-data#37](https://github.com/e-mission/e-mission-eval-private-data/pull/37)
>> 
>> You should probably resolve those first, and get that PR merged before refactoring.


The two functions 'get_distance_matrix' and 'single_cluster_purity' required here are not affected  in that PR and will remain unaffected in future.

Reply:
However, as mentioned we'll have to wait for https://github.com/e-mission/e-mission-eval-private-data/pull/37  to be merged before we begin this movement.